javascript/javascript2/week2/index.js

```diff
@@ -0,0 +1,112 @@
+let numbers = [1, 2, 3, 4];
+let newNumbers = [];
+
+for (let i = 0; i < numbers.length; i++) {
+  if (numbers[i] % 2 !== 0) {
+    newNumbers[i] = numbers[i] * 2;
+  }
+}
+
+numbers.forEach((num) => {
+  if (num % 2 !== 0) {
+    newNumbers.push(num * 2);
+  }
+});
+
+const oddNums = numbers.filter((num) => num % 2 !== 0);
+const oddTimes2 = oddNums.map((odd) => odd * 2);
+console.log(oddNums);
+console.log(oddTimes2);
+newNumbers.push(oddTimes2);
+console.log(newNumbers);
+
+// Create an array of movies containing the movies with a short title (you define what short means)
+const shortTitles = movies
+  .filter((titles) => {
+    movies.forEach((x) => x.title);
+    return titles.title.length < 5;
+  })
+  .map((x) => x.title);
+console.log(shortTitles);
+
+// Create an array of movie titles with long movie titles
+
+const longTitles = movies
+  .filter((titles) => {
+    movies.forEach((longT) => longT.title);
+    return titles.title.length > 5;
+  })
+  .map((x) => x.title);
+console.log(longTitles);
+
+// Count the number of movies made between 1980-1989 (including both the years)
+
+const numOfMovies = movies
+  .filter((a) => {
+    if (a.year >= 1980 && a.year <= 1989) {
+      return a;
+    }
+  })
```

Comment:
Here it can be
`const numOfMovies = movies.filter((movie) => movie.year >= 1980 && movie.year <= 1989)`

It is also good to use meaningful names instead of a, b,etc..